### PR TITLE
Feature/multi uploads xmlrpc

### PR DESCRIPTION
**Pull request by mzorz**
Provides multiupload-multicancel capabilties and tests on FluxC for XMLRPC.

cc @aforcier

### CODE REVIEW

**Comment by aforcier** [comment]
Interesting, the cancellation test fails for me even though it has the [same timeout](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/401/files#diff-89d0e5351b721034c8f85ffb620a4471R441) as the WP.com tests.

It works if I step the wait time down to `100ms`, but I think I'll just use a slightly larger test image in the tests (`9KB` is maybe excessively small 😂). No need to change it.

**Comment by mzorz** [comment]
interesting! For me I had to boost the timeout to 60 seconds just for the single image upload and a ~500kb image I use for testing
These tests might benefit from having a dataset attached to them

**Comment by aforcier** [comment]
> interesting! For me I had to boost the timeout to 60 seconds just for the single image upload and a ~500kb image I use for testing
These tests might benefit from having a dataset attached to them

A set different media for each test to run through 🤔 that might be really neat to have.

File: example/src/androidTest/java/org/wordpress/android/fluxc/release/ReleaseStack_MediaTestXMLRPC.java [hunk 1]
```diff
@@ -42,6 +44,9 @@
         FETCHED_MEDIA,
         DELETED_MEDIA,
         UPLOADED_MEDIA,
+        UPLOADED_MUTIPLE_MEDIA, // these don't exist in FluxC, but are an artifact to wait for all
```
Comment (aforcier) [inline example/src/androidTest/java/org/wordpress/android/fluxc/release/ReleaseStack_MediaTestXMLRPC.java]: `MULTIPLE` is missing an `L` in several places.

Comment (mzorz) [inline example/src/androidTest/java/org/wordpress/android/fluxc/release/ReleaseStack_MediaTestXMLRPC.java]: fixed in a8443cb

**Comment by aforcier** [comment]
:shipit:

**Pull request merged**